sdk/servicebus/service-bus/test/backupMessageSettlement.spec.ts

```diff
@@ -45,503 +50,285 @@ describe("Backup message settlement - Through ManagementLink", () => {
     deadLetterReceiver = serviceBusClient.test.createDeadLetterReceiver(entityNames);
   }
 
-  function afterEachTest(): Promise<void> {
-    return serviceBusClient.test.afterEach();
+  afterEach(async () => {
+    await serviceBusClient.test.afterEach();
+  });
+
+  async function sendReceiveMsg(testMessages: ServiceBusMessage): Promise<ReceivedMessageWithLock> {
+    await sender.sendMessages(testMessages);
+    const msgs = await receiver.receiveMessages(1);
+
+    should.equal(Array.isArray(msgs), true, "`ReceivedMessages` is not an array");
+    should.equal(msgs.length, 1, "Unexpected number of messages");
+    should.equal(msgs[0].body, testMessages.body, "MessageBody is different than expected");
+    should.equal(msgs[0].messageId, testMessages.messageId, "MessageId is different than expected");
+    should.equal(msgs[0].deliveryCount, 0, "DeliveryCount is different than expected");
+
+    return msgs[0];
+  }
+
+  async function testComplete(): Promise<void> {
+    const testMessages = entityNames.usesSessions
+      ? TestMessage.getSessionSample()
+      : TestMessage.getSample();
+    const msg = await sendReceiveMsg(testMessages);
+    await receiver.close();
+    let errorWasThrown = false;
+    try {
+      await msg.complete();
+    } catch (err) {
+      should.equal(
+        err.message,
+        `Failed to ${DispositionType.complete} the message as the AMQP link with which the message was received is no longer alive.`,
+        "Unexpected error thrown"
+      );
+      errorWasThrown = true;
+    }
+
+    receiver = await serviceBusClient.test.getPeekLockReceiver(entityNames);
+    if (entityNames.usesSessions) {
+      should.equal(errorWasThrown, true, "Error was not thrown for messages with session-id");
+      const msgBatch = await receiver.receiveMessages(1);
+      await msgBatch[0].complete();
+    } else {
+      should.equal(errorWasThrown, false, "Error was thrown for sessions without session-id");
+    }
+    await testPeekMsgsLength(receiver, 0);
   }
 
-  describe("Operations On Messages After Receiver Is Closed", function(): void {
-    afterEach(async () => {
-      await afterEachTest();
-    });
+  it(noSessionTestClientType + ": complete() removes message", async function(): Promise<void> {
```

Comment:
What happened to 
`[noSessionTestClientType, withSessionTestClientType].forEach((testClientType) => {` ?

Reply:
I had that format at one point, but moved away from it to keep things simple.
It helps with debugging or running individual tests.

For example, if I had to run/debug 1 test, I would add the `.only`
If I use the format you are suggesting here, the same test would run twice.
Now, I have to update the array to have a single entity type in order to run just the one test.

So, making things easier for people to use the tests trumped the need to optimize the code itself.